### PR TITLE
feat: selenium standalone for UI testing

### PR DESCRIPTION
**Pull request by mkolasinski-splunk**
This PR introduces execution of UI tests on selenium standalone instance. This is achieved by usage of `SELENIUM_HOST` env var.
Changes:

- introduction of selenium standalone UI tests execution
- fixes for failing test cases, in most cases `self.wait_to_be_clickable("element")`
- improved `toggle()` of checkbox component
- fixes for unit tests

BREAKING CHANGE: Default UI tests executor changed from Saucelabs to Selenium standalone

### CODE REVIEW

**Review by mbruzda-splunk — APPROVED** [review]

File: README.md
Comment (artemrys) [inline README.md:1]: Should we have a change for "Table of contents" in this PR?

File: pytest_splunk_addon_ui_smartx/base_test.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def __init__(
         debug=False,
         cred=("admin", "Chang3d!"),
         headless=False,
+        saucelabs=False,
```
Comment (artemrys) [inline pytest_splunk_addon_ui_smartx/base_test.py:63]: `saucelabs` variable is not being used in this PR? should we remove it?

File: pytest_splunk_addon_ui_smartx/base_test.py [hunk 2]
```diff
@@ -83,6 +87,17 @@ def __init__(
...
                     )
+                elif selenium_host:
+                    options_firefox = webdriver.FirefoxOptions()
+                    options_firefox.add_argument("--ignore-ssl-errors=yes")
+                    options_firefox.add_argument("--ignore-certificate-errors")
+                    options_firefox.add_argument("--disable-dev-shm-usage")
+
+                    self.browser = webdriver.Remote(
+                        command_executor=f"{selenium_host}:4444/wd/hub",
+                        options=options_firefox,
```
Comment (artemrys) [inline pytest_splunk_addon_ui_smartx/base_test.py:98]: There were much more options used in the Saucelabs-based solution (including screen size and other options). How screen size is being controlled with Selenium-based solution?

File: pytest_splunk_addon_ui_smartx/plugin.py [hunk 3]
```diff
@@ -156,7 +177,7 @@ def get_browser_scope(fixture_name, config):
     """
     Set the scope of the browser dynamically.
     """
-    if config.getoption("--local") and config.getoption("--persist-browser"):
```
Comment (artemrys) [inline pytest_splunk_addon_ui_smartx/plugin.py:159]: Is this change necessary with this PR?

File: pytest_splunk_addon_ui_smartx/plugin.py [hunk 4]
```diff
@@ -137,9 +148,18 @@ def ucc_smartx_configs(request):
     else:
         headless_run = False
 
+    if request.config.getoption("--saucelabs"):
+        saucelabs_run = True
+        LOGGER.debug("--saucelabs")
+    else:
+        saucelabs_run = False
```
Comment (artemrys) [inline pytest_splunk_addon_ui_smartx/plugin.py:155]: I believe you wanted to remove this section on today's meeting.

**Review by artemrys — CHANGES_REQUESTED** [review]

File: README.md
Comment (mkolasinski-splunk) [reply README.md:1]: Table of contents gets automatically updated in my IDE when saving the file.

File: pytest_splunk_addon_ui_smartx/base_test.py (hunk 2 above)
Comment (artemrys) [reply pytest_splunk_addon_ui_smartx/base_test.py:98]: You are right, should we add `screenResolution` to our new configuration?

Comment (mkolasinski-splunk) [reply pytest_splunk_addon_ui_smartx/base_test.py:98]: Default resolution is [1360x1020](https://github.com/SeleniumHQ/docker-selenium/blob/trunk/README.md#setting-screen-resolution) and can be changes via env vars when instantiating selenium container.
I don't think we need to bother about that without any specific requirements.
If needed, can be adjusted on k8s-manifests.

File: pytest_splunk_addon_ui_smartx/base_test.py (hunk 1 above)
Comment (mkolasinski-splunk) [reply pytest_splunk_addon_ui_smartx/base_test.py]: removed

**Review by artemrys — APPROVED** [review]

**Review by artemrys — APPROVED** [review]
